src/main/kotlin/ecommerce/model/CartProduct.kt

```diff
@@ -0,0 +1,31 @@
+package ecommerce.model
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.Table
+import jakarta.persistence.UniqueConstraint
+
+@Entity
+@Table(
+    name = "cart_products",
+    uniqueConstraints = [UniqueConstraint(columnNames = ["cart_id", "option_id"])],
+)
+class CartProduct(
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "cart_id", nullable = false)
+    var cart: Cart,
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "option_id", nullable = false)
+    var option: Option,
+    @Column(name = "quantity", nullable = false)
+    var quantity: Int = 1,
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    val id: Long = 0L,
+)
```

Comment:
```suggestion
class CartProduct(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cart_id", nullable = false)
    var cart: Cart,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "option_id", nullable = false)
    var option: Option,

    @Column(name = "quantity", nullable = false)
    var quantity: Int = 1,
)
```

I personally prefer keeping `id` field on the very top, since that (to me) is the most important value when it comes to entity.

Also, when there's multiple annotations added, I like to keep some line breaks in between fields, as it helps increase the readability. (this is also just my opinion 😂 )

Reply:
I actually keep the id field at the end for a practical reason:
in Kotlin, if the first constructor parameter has a default value (0L), you still need to use named arguments for all the following parameters when creating the object.
For example, if `id` were first, creating an `Option` in tests would look like this:
```
Option(
    id = 0L,
    name = "test",
    price = 10.2,
    quantity = 52,
    imageUrl = "http://localhost:8080/image/upload/product1.png",
)
```
By keeping `id` at the end, I can just rely on its default value and create test much more cleanly:
```
Option(
    "test",
    10.2,
    52,
    "http://localhost:8080/image/upload/product1.png",
)
```
It also feels more natural for me to read the business fields first, and the technical id last. 😄

But I get your point about readability - I’ve now added some line breaks between fields with multiple annotations to make it easier to scan visually.
Thanks for the tip!

Reply:
Justin actually showed us this Kotlin trick in the lecture - if you move the id to the end with a default value, creating test  becomes so much cleaner.

I used to always keep `id` first, but after that lecture I went through and moved all `id`s to the end.😅

Reply:
I use KtlinFormat to keep the code neat, and it removes empty lines. I need to use it to pass the Check validation. So, I can’t leave empty lines for separation if I want the Check to pass.🤣

Reply:
Great! Since this is a suggestion, I understand your point and keeping it as is looks fine 👍